### DISCUSSION TITLE
Allow variables with hardcoded values

### DISCUSSION THREAD

**Discussion by gabzim** [grafana/grafana#38860] [Dashboards, Panels & Visualizations]
I haven't been able to pull this off if it exists already so I apologize in advance if that's the case. My use case is simple, I have a dashboard that has a set of variables. Most of them I want to load dynamically from my datasources, but there are a couple of variables that I'd like to hardcode since they will rarely ever change.

My ask would be, in addition to being able to query a datasource to retrieve the values of each variable, I'd like to be able to specify them explicitly and have that data live in the dashboard (similarly to how you would add tags in a dashboard?). This way I'll save unnecessary queries to my backend and the variables will also load instantly.

**Comment by allomorphy** [comment]
@gabzim 
Maybe not a perfect fit for your use case, but you can define one or more static values for a variable by selecting the variable `Type` of `Custom` in the dropdown as shown here 👇 
<img width="1121" alt="image" src="https://user-images.githubusercontent.com/50141922/132116184-2899e00e-8dbf-4f38-a1a7-6a437d1b22c2.png">

**Comment by gabzim** [comment]
I feel so dumb for missing this, thanks @allomorphy 👍  I really appreciate the help.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
